main.go

```diff
@@ -45,4 +55,15 @@ func main() {
 	close(swapIntentOperationBroadcastChan)
 	close(solverOperationSubmitChan)
 	close(governanceSignatureChan)
+
+	// Questions:
+	// entities:
+	//    backend - is this the bundler that gets the intent, distribute to solvers, collect solutio and create the metaTransactions? you can it backend and not bundler as there can be other configurations?
+	//    governance - what is this entity? is it part of the dApp or part of Atlas?
+	//    user - represents a user application or a real user + dApp ui?
+	//
+	// smart contracts - can you make the 9 smart contract to their "owner" - dApp, Atlas, system?
+	// looks like simulator is not in use
+	// what is txBuilder? provided by Atlas or dApp? if by atlas, why a separate contract?
+	// why there is a need for SimpleRFQSolver?
```

Comment:
In this example, SimpleRFQSolver is the contract of the solver. Any solver would need to deploy a contract of their own, to handle their operations solving the intents, pay their bids, and any other steps needed for them to complete their solution.

Reply:
again, not sure why solvers need contract. Can't they just use their wallet?

Reply:
In the simple scenario where the user intent is to buy X amount Y, and a solver happen to have this amount in their possession and willing to sell it, then yes, I guess a simple transaction would work.

We are expecting solvers to run more elaborate strategies to solve intents, which require a smart contract.

Reply:
There are a few reasons for solvers to use contracts:
1) Solver contracts can inherit our SolverBase contract, which means they only code their custom logic which captures the mev. All the other complexity gets abstracted away, such as paying bids and safety checks
2) In this super simple case, it might be possible to do via a solver wallet, but it assumes they will have funds and pre-approve the execution env contract to take those funds from that wallet (a bad idea for security but lets ignore that). A wallet cannot execute more complex searcher mev strategies, such as fulfilling the intent by comparing the arbitrage across Uniswap, 1inch, and Curve, and then swapping via the best price DEX, capturing the arb profit.

So solvers need to be contracts to enable more complex mev strategies and to add some security